### ISSUE TITLE
Docker image - Please include zope directly when building image

### ISSUE THREAD

**Issue by thomaslc66** [NginxProxyManager/nginx-proxy-manager#2997]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes 
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**
Zope not installed by default in the docker image, meaning that each restart we need to do a pip install zope

**Nginx Proxy Manager Version**
:latest

**Expected behavior**
pip install zope should be done and included in the already existing docker image.
The docker image should be totally working without having to install additional packages.

**Comment by ipedrazas** [comment]
The bit that worries me the most is not that you think that zope should be installed by default, what worries me is that you install zope after each restart...

Have you thought about creating your own Docker image? 

```
FROM  jc21/nginx-proxy-manager:latest
RUN pip install zope
```

**Comment by thomaslc66** [comment]
Whooooaaa, what a condescending way to answer.

The real question is why, if it's something that is required in order to renew certificates, and if there are multiple issues raised about it, it's not already done in the initial docker file. Plus I'm using that in an Unraid server and creating an image would work but this would mean having to update everything manually when Unraid offers a way to automatically update as new versions are published and I don't know how you deal with your proxy server but for me, it's up and running most of the time, so there's no "reboot" except maybe once or twice a year.

**Comment by SignFinder** [comment]
> Whooooaaa, what a condescending way to answer.

Might @ipedrazas answered by this way, because your question looks impolite?
You trying to waste the time of another people and NPM developer to solve your trivial and personal problem, what is not related to NPM.
And after the answer - how to solve your problem by such standard and ordinary way - you continue to steal others time.

**Comment by thomaslc66** [comment]
@SignFinder Let's agree to disagree.

I will not take into consideration that you think I'm my problem is trivial and personal because you clearly have done no research with your precious time. There are multiple issues related to this package in this repository. If for you the resolution of all those issues is "Create your own docker image" it means that all those people have to do it and if all those people have to do it, why not do it directly in the main repository and help everybody at the same time? 

If you were using your time correctly you would see that this is impacting a lot of people:
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/2825
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/3015
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/2994
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/2678
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/2709

There's even pull request of people trying to help:
https://github.com/NginxProxyManager/nginx-proxy-manager/pull/2756

So I don't think I'm stealing people's time and clearly not NPM developer as pip is related to python and NPM to node/js, but clearly, you are not using yours to help.

Have a nice one

**Comment by jc21** [comment]
FWIW I'm not putting much if any attention into the v2 codebase, particularly with (and because of) these certbot plugin issues and never-ending python problems. V3 will Acme.sh instead of Certbot.

**Comment by DuvelCorp** [comment]
> > Whooooaaa, what a condescending way to answer.
> 
> Might @ipedrazas answered by this way, because your question looks impolite? You trying to waste the time of another people and NPM developer to solve your trivial and personal problem. And after the answer - how to solve your problem by such standard and ordinary way - you continue to steal others time.

How can you state that this is a trivial problem.? And also where di you see that he was being impolite.
The whole paradigm of containerization is that any component need by any other component in your container, should be bundled in the image. Having to manually install things needed by a major feature (ie handling certs) on container is an abomination. And the issue is there for > 1 year so why is it not in the container yet???

**Comment by qdrop17** [comment]
> FWIW I'm not putting much if any attention into the v2 codebase, particularly with (and because of) these certbot plugin issues and never-ending python problems. V3 will Acme.sh instead of Certbot.

I genuinely appreciate this project, but isn't it a bit problematic for the trust (in this project) if such fixes / PRs don't get merged (even for older codebases). Because the issue with the custom Dockerfile is that it's quite cumbersome to push updates to users (such as security updates via Watchtower) which is also very import for the trust in an opensource project - especially if it's often an internet-facing service.

I didn't find any issues that come with installing zope despite the image gaining couple of MB.

**Comment by Paillat-dev** [comment]
Yes. I am in a position where I am unable to use this project with docker swarm UNLESS I rebuild the image and push it to my own registry. For a dumb dependency issue that can be fixed in 2 lines.

**Comment by Paillat-dev** [comment]
Not wanting to advertise, just to be helpful with anyone here, an image with zope installed can be found here `ghcr.io/paillat-dev/npm-with-dependencies:main`

**Comment by qdrop17** [comment]
yes, same here: no advertising, but https://github.com/ZoeyVid/NPMplus is an additional option with some additional security features baked in.
